csrc/logical_domain_map.cpp

```diff
@@ -693,11 +693,11 @@ bool ComputeAtLogicalDomainMap::canMap(
     const TensorDomain* td_b,
     const IterDomain* id_b) const {
   NVF_ERROR(
-      id_a->definition() == nullptr || id_a->isRFactorProduct(),
```

Comment:
Previously we assumed a root domain has no definition, which may not be the case anymore.